webapp/src/pages/EnableTwoFactorAuthPage/EnableTwoFactorAuthPage.tsx

```diff
@@ -0,0 +1,91 @@
+import './EnableTwoFactorAuthPage.css';
+import { useState } from 'react';
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  Loading,
+} from '../../shared/components';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import { TWO_FACTOR_AUTH_QR_EP_URL, USER_ME_URL } from '../../shared/urls';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { authApi } from '../../shared/services/ApiService';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import { useNavigate } from 'react-router-dom';
+import { ResponseError } from '../../shared/generated';
+
+export default function EnableTwoFactorAuthPage() {
+  const { authUser, setAuthUser, isLoading } = useAuth();
+  const { goBack } = useNavigation();
+  const [code, setCode] = useState('');
+  const { warn, notify } = useNotificationContext();
+  const navigate = useNavigate();
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
+    event,
+  ) => {
+    event.preventDefault();
+    if (!authUser) {
+      return;
+    }
+    try {
+      await authApi.authControllerEnableTwoFactorAuthentication({
+        twoFactorAuthenticationCodeDto: { code },
+      });
+      setAuthUser({ ...authUser, isTwoFactorAuthenticationEnabled: true });
+      notify('You enabled 2FA');
+      navigate(USER_ME_URL, { replace: true });
+    } catch (error: unknown) {
+      if (error instanceof ResponseError) {
+        const responseBody = await error.response.json();
+        if (responseBody.message) {
+          warn(responseBody.message);
+        } else {
+          warn(error.response.statusText);
+        }
+      } else if (error instanceof Error) {
+        warn(error.message);
+      } else {
+        warn('Could not enable 2FA');
+      }
+    }
+  };
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  if (!authUser || authUser.isTwoFactorAuthenticationEnabled) {
+    return <NotFoundPage />;
+  }
+
+  return (
+    <div className="enable-two-factor-auth-page">
+      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        enable 2fa
+      </Header>
+      <div className="enable-two-factor-auth-page-main">
+        <img
+          className="enable-two-factor-auth-page-qrcode"
+          src={TWO_FACTOR_AUTH_QR_EP_URL}
+          alt="QR Code"
+        />
```

Comment:
I have a similar issue related to what @Damrod explained. In my case, when disabling and enabling again 2FA I got the same QR as before (when a new one was expected) and I could enable 2FA.

After investigating a bit I think the proble is here, the QR image is being cached by the browser because you are requesting it as an asset. So, when you leave the page and re-enter (after a new 2FA activation, even from another user) the browser returns the old cached image which is tied to the previous session.

Reply:
One solution to fix this could be to attach a query string to the url with the current utc time. This way we would have a different url each time the user requests the QR and the browser would be forced to request the resource again (https://stackoverflow.com/questions/126772/how-to-force-a-web-browser-not-to-cache-images)

The downside for this option is the browser would be caching assets that the user is not going to consume again. But since the user is not going to visit this page frequently, not a big deal.

Reply:
But I see that a more elegant solution would be to directly append extra headers to the controller response to indicate the browser we don't want to cache the resource, such as `@Header('Cache-Control', 'no-cache')`

Reply:
Thank you, guys! I added the query string because the header didn't work for me.

Reply:
Thanks! I am investigating this because it neither worked for me. I think it's due to client side navigation. I'll write my conclusions in this thread 😄

Reply:
The current fix is still not working for me, guys. Images below. Different query string, same image:
![Screenshot from 2023-01-19 13-05-27](https://user-images.githubusercontent.com/2774420/213426684-455b78bc-a15a-462f-a88f-0a8ac1f4c64d.png)
![Screenshot from 2023-01-19 13-05-16](https://user-images.githubusercontent.com/2774420/213426686-83a335d6-2685-4ed2-b494-0d0630cd104b.png)

Reply:
steps to reproduce?

Reply:
1 - login with user `a`
2 - click enable 2fa
2bis - open qr code in a different tab
3 - scan qr code from original tab with cellphone app
4 - log out
5 - log in with 'a', check that 2fa works as expected
6 - log out from 'a'
7 - log in with 'b'
8 - try to enable 2fa clicking the button
9 - see that prompted qr code is the same than the one in the side tab i opened in point 2bis

Reply:
I've rebuilt the app and the volumes and now it works.

??